### DISCUSSION TITLE
Advanced hunk actions

### DISCUSSION THREAD

**Discussion by carrascomj** [helix-editor/helix#4778] [Ideas] [answered]
I have been using the git gutter PR #3890 for a while and it is great.

Apart from the functionality to **jump between hunks** (#4650), the thing that I miss from [gitsigns.nvim](https://github.com/lewis6991/gitsigns.nvim) is the ability to **stage hunks**. This is like running `git add` but only for the modified lines under the selection (when running `:Gitsigns stage_hunk`, similar in concept to `git add -p`). There is a lot of other extra functionality coming that plugin but I don't really use it.

Is there interest in an implementation of this? Is anyone working on this? I put this under discussions instead of issues because I don't know if there are plans to move the Git/VCS part to its own plugin once the plugin system is arranged in the future. Even if it gets implemented in the core forever, maybe these extra features based on VCS integration will be delegated to plugins. Are there plans about this?

> For reference, I'll leave here the lua implementation of [stage_hunk in gitsigns.nvim](https://github.com/lewis6991/gitsigns.nvim/blob/fa1e8c589a7c3a7fcadc89bf0d2bc2b7f6e6c3be/lua/gitsigns/hunks.lua#L56).

**Comment by carrascomj** [comment, marked as answer]
This was implemented in https://github.com/helix-editor/helix/pull/5736 and will be further improved in https://github.com/helix-editor/helix/pull/10178.

**Reply by Sirius902** [reply]
If I'm understanding those PRs only deal with the resetting (un-staging of hunks) but does not seem to cover staging hunks. Personally I'm still interested in being able to stage hunks under the cursor(s) as well.
